src/components/learner-credit-management/cards/BaseCourseCard.jsx

```diff
@@ -0,0 +1,100 @@
+import React from 'react';
```

Comment:
Largely duplicate of the original `CourseCard` component, but renamed under `BaseCourseCard` and allows devs to override the footer actions depending on the use case the card is rendered (in some places, we may not want the actions to appear but want to otherwise re-use the `CourseCard`).

By abstracting out `CourseCard` to a more flexible `BaseCourseCard` allows `CourseCard` to rely on `NewAssignmentModalButton`, which itself relies on `BaseCourseCard`. Without the introduction of `BaseCourseCard`, there would be an import dependency cycle when `NewAssignmentModalButton` tries to render `CourseCard`.